cub/cub/agent/agent_find.cuh

```diff
@@ -0,0 +1,253 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#include <cub/config.cuh>
+
+#include <cub/iterator/cache_modified_input_iterator.cuh>
+#include <cub/thread/thread_load.cuh>
+#include <cub/util_arch.cuh>
+#include <cub/util_type.cuh>
+
+#include <cuda/std/__type_traits/integral_constant.h>
+
+CUB_NAMESPACE_BEGIN
+namespace detail::find
+{
+template <int NominalBlockThreads4B,
+          int NominalItemsPerThread4B,
+          typename ComputeT,
+          int VectorLoadLength,
+          CacheLoadModifier _LOAD_MODIFIER,
+          typename ScalingType = cub::detail::MemBoundScaling<NominalBlockThreads4B, NominalItemsPerThread4B, ComputeT>>
+struct agent_find_policy : ScalingType
+{
+  // Number of items per vectorized load
+  static constexpr int vector_load_length = VectorLoadLength;
+
+  // Cache load modifier for reading input elements
+  static constexpr CacheLoadModifier load_modifier = _LOAD_MODIFIER;
+};
+
+template <typename agent_find_policy_t, typename InputIteratorT, typename OutputIteratorT, typename OffsetT, typename ScanOpT>
+struct agent_t
+{
+  // The input value type
+  using InputT = typename ::cuda::std::iterator_traits<InputIteratorT>::value_type;
+
+  // Vector type of InputT for data movement
+  using VectorT = typename CubVector<InputT, agent_find_policy_t::vector_load_length>::Type;
+
+  // Input iterator wrapper type (for applying cache modifier)
+  // Wrap the native input pointer with CacheModifiedInputIterator
+  // or directly use the supplied input iterator type
+  using WrappedInputIteratorT =
+    ::cuda::std::_If<::cuda::std::is_pointer<InputIteratorT>::value,
+                     CacheModifiedInputIterator<agent_find_policy_t::load_modifier, InputT, OffsetT>,
+                     InputIteratorT>;
+
+  static constexpr int BLOCK_THREADS      = agent_find_policy_t::BLOCK_THREADS;
+  static constexpr int ITEMS_PER_THREAD   = agent_find_policy_t::ITEMS_PER_THREAD;
+  static constexpr int TILE_ITEMS         = BLOCK_THREADS * ITEMS_PER_THREAD;
+  static constexpr int vector_load_length = ::cuda::std::min(ITEMS_PER_THREAD, agent_find_policy_t::vector_load_length);
```

Comment:
Suggestion: it seems we're mixing styles between `BLOCK_THREADS` and `vector_load_length` (caps vs no caps). I'm not exactly sure which one we use.

Reply:
guh.... I  missed in my previous push. should all be lower case